### PR TITLE
Fix BC Free for 2p problems

### PR DESCRIPTION
**Pull request by totto82**
This fixes issue when using BCPROP with type FREE and DIRICHLET in a 2p run.

### CODE REVIEW

File: opm/models/blackoil/blackoillocalresidualtpfa.hh [hunk 1]
```diff
@@ -493,8 +493,9 @@ public:
         OPM_TIMEBLOCK_LOCAL(computeBoundaryFluxFree);
         std::array<short, numPhases> upIdx;
         std::array<short, numPhases> dnIdx;
-        RateVector volumeFlux;
-        RateVector pressureDifference;
+        Evaluation volumeFlux[numPhases];
+        Evaluation pressureDifference[numPhases];
```
Comment (bska) [inline opm/models/blackoil/blackoillocalresidualtpfa.hh:497]: Why built-in arrays?  I can't think of a reason we would prefer this to `std::array<>`.

**Review by bska — COMMENTED** [review]
I want both @atgeirr and @hnil to have a look at the changes to the TPFA lineariser code.  The only thing that stands out to me though, is using built-in arrays over `std::array<>` in the residual impliementation.

File: opm/models/blackoil/blackoillocalresidualtpfa.hh (hunk 1 above)
Comment (totto82) [reply opm/models/blackoil/blackoillocalresidualtpfa.hh:497]: I just copied what was used in eclfluxmodule.hh. I can change to `std:.array<Evaluation, numPhases>`

File: opm/models/blackoil/blackoillocalresidualtpfa.hh [hunk 2]
```diff
@@ -574,8 +575,8 @@ public:
                                                                              inIdx,
                                                                              alpha,
                                                                              bdyInfo.exFluidState);
+            EnergyModule::addHeatFlux(bdyFlux, heatFlux);
```
Comment (atgeirr) [inline opm/models/blackoil/blackoillocalresidualtpfa.hh:578]: I think this change is correct, but then the heatFlux variable should also be created inside the `if constexpr(enableEnergy) {}` block?

**Review by atgeirr — APPROVED** [review]
I agree with the fix, just a minor adjustment asked for.

File: opm/models/blackoil/blackoillocalresidualtpfa.hh (hunk 2 above)
Comment (totto82) [reply opm/models/blackoil/blackoillocalresidualtpfa.hh:578]: Good point. Will fix.

**Comment by totto82** [comment]
jenkins build this please

**Comment by atgeirr** [comment]
Thanks, merging!

**Pull request merged**